src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -3046,3 +3057,232 @@ citus_internal_update_relation_colocation(PG_FUNCTION_ARGS)
+			appendStringInfoString(insertColocationCommand, "0");
+		}
+	}
+	else
+	{
+		appendStringInfoString(insertColocationCommand, "0");
+	}
+
+	appendStringInfoString(insertColocationCommand, ", ");
+
+	if (distributionColumnCollation != InvalidOid)
+	{
+		/* would be great if we could use regcollation, but it's not available on PG12 */
+
+		Datum collationIdDatum = ObjectIdGetDatum(distributionColumnCollation);
+		HeapTuple collationTuple = SearchSysCache1(COLLOID, collationIdDatum);
+
+		if (HeapTupleIsValid(collationTuple))
+		{
+			Form_pg_collation collationform =
+				(Form_pg_collation) GETSTRUCT(collationTuple);
+			char *collationName = NameStr(collationform->collname);
+			char *collationSchemaName = get_namespace_name(collationform->collnamespace);
+
+			appendStringInfo(insertColocationCommand,
+							 "(select oid from pg_collation"
+							 " where collname = %s"
+							 " and collnamespace = %s::regnamespace)",
+							 quote_literal_cstr(collationName),
+							 quote_literal_cstr(collationSchemaName));
+		}
+		else
+		{
+			/* pg_dist_colocation contains an invalid collation, just insert 0 */
+			appendStringInfoString(insertColocationCommand, "0");
+		}
+
+		ReleaseSysCache(collationTuple);
+	}
+	else
+	{
+		appendStringInfoString(insertColocationCommand, "0");
+	}
+
+	appendStringInfoString(insertColocationCommand, ")");
+
+	return insertColocationCommand->data;
+}
+
+
+/*
+ * SyncDeleteColocationGroupToNodes deletes a pg_dist_colocation record from workers.
+ */
+void
+SyncDeleteColocationGroupToNodes(uint32 colocationId)
+{
+	char *command = ColocationGroupDeleteCommand(colocationId);
+
+	/*
+	 * We require superuser for all pg_dist_colocation operations because we have
+	 * no reasonable way of restricting access.
+	 */
+	SendCommandToWorkersWithMetadataViaSuperUser(command);
+}
+
+
+/*
+ * ColocationGroupCreateCommand returns a command for creating a colocation group.
+ */
+static char *
+ColocationGroupDeleteCommand(uint32 colocationId)
+{
+	StringInfo deleteColocationCommand = makeStringInfo();
+
+	appendStringInfo(deleteColocationCommand,
+					 "SELECT pg_catalog.citus_internal_delete_colocation_metadata(%d)",
+					 colocationId);
+
+	return deleteColocationCommand->data;
+}
+
+
+/*
+ * ColocationGroupCreateCommandList returns the full list of commands for syncing
+ * pg_dist_colocation.
+ */
+List *
+ColocationGroupCreateCommandList(void)
+{
+	List *commandList = NIL;
+
+	Relation pgDistColocation = table_open(DistColocationRelationId(), AccessShareLock);
+
+	bool indexOK = false;
+	SysScanDesc scanDescriptor = systable_beginscan(pgDistColocation, InvalidOid, indexOK,
+													NULL, 0, NULL);
+
+	HeapTuple colocationTuple = systable_getnext(scanDescriptor);
+
+	while (HeapTupleIsValid(colocationTuple))
+	{
+		Form_pg_dist_colocation colocationForm =
+			(Form_pg_dist_colocation) GETSTRUCT(colocationTuple);
+
+		char *command =
+			ColocationGroupCreateCommand(colocationForm->colocationid,
```

Comment:
at the moment not really because we only create a new co-location group when using a new shard count / distribution column type combination and not co-locating with another table